src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilderResource.java

```diff
@@ -138,12 +138,18 @@ public boolean equals(final Object o)
         {
             final BuildingBuilderResource that = (BuildingBuilderResource) o;
 
-            return (this.getAvailable() == that.getAvailable() && this.getPlayerAmount() == that.getPlayerAmount());
+            return this.getAvailable() == that.getAvailable() && this.getPlayerAmount() == that.getPlayerAmount();
         }
 
         return false;
     }
 
+    @Override
+    public int hashCode()
+    {
+        return super.hashCode() + amountAvailable + amountPlayer;
```

Comment:
Because hashcodes should be unique-ish and shouldn't collide very often. Anytime amountAvailable + amountPlayer equal the same thing you will have a hash collision. There have been plenty of studies on optimial hash functions, and one of the simplest ones is just multiplying everything by 31 at each step, so in this case you would do:

31*(31*super.hashCode() + amountAvailable) + amountPlayer


Sorry for any formatting issues, on mobile.